R/run_pipeline.R

```diff
@@ -1,98 +1,47 @@
-#' Run machine learning pipeline
+#' Run the machine learning pipeline
 #'
-#' @param dataset TODO
-#' @param method TODO
-#' @param outcome_colname TODO
-#' @param outcome_value TODO
-#' @param hyperparameters TODO
-#' @param metric TODO
-#' @param permute TODO
-#' @param nfolds fold number for cross-validation
+#' @param dataset dataframe with an outcome variable and other columns as features
+#' @param method ML method ("regLogistic", "svmRadial", "rpart2", "rf", "xgbTree")
+#' @param outcome_colname column name as a string of the outcome variable
+#' @param outcome_value outcome value of interest as a string
+#' @param hyperparameters dataframe of hyperparameters (default: default_hyperparams)
+#' @param permute run permutation imporance (default: FALSE)
+#' @param nfolds fold number for cross-validation (default: 5)
+#' @param training_frac fraction size of data for training (default: 0.8)
 #' @param seed random seed (default: NA)
 #'
 #' @return named list with results
 #' @export
 #' @author Begüm Topçuoğlu, \email{topcuoglu.begum@@gmail.com}
+#' @author Zena Lapp, \email{zenalapp@@umich.edu}
+#' @author Kelly Sovacool, \email{sovacool@@umich.edu}
 #'
 run_pipeline <-
   function(dataset,
            method,
            outcome_colname = NA,
            outcome_value = NA,
            hyperparameters = mikRopML::default_hyperparams,
-           metric = "ROC",
            permute = FALSE,
-           nfolds = 5,
+           nfolds = as.integer(5),
+           training_frac = 0.8,
            seed = NA) {
-    if (!is.na(seed)) {
-      set.seed(seed)
-    }
-
-    methods <- c("regLogistic", "svmRadial", "rpart2", "rf", "xgbTree")
-    if (!(method %in% methods)) {
-      stop(paste0(
-        "Method '",
-        method,
-        "' is not supported. Supported methods are:\n    ",
-        paste(methods, collapse = ", ")
-      ))
-    }
 
+    # input validation functions
+    check_method(method)
+    check_dataset(dataset)
+    validate_permute(permute)
+    validate_nfolds(nfolds, dataset)
+    validate_training_frac(training_frac)
+    validate_seed(seed)
```

Comment:
Should we put this in a wrapper function?

Reply:
Sure! I also renamed `validate` to `check` for brevity and consistency
e054e6ee5a75f0b784ca00a5186becac911726ec